### ISSUE TITLE
long expires time

### ISSUE THREAD

**Issue by KonoromiHimaries** [gorhill/uBlock#3406]
most users have a old filters
and so happened
`https://www.wykop.pl/link/4094131/brak-mozliwosci-zalogowania-do-mbanku-z-wlaczonym-ublockiem/`

filters have min. 2 days expiries time
![2018-01-05 23_01_50-nano filter viewer](https://user-images.githubusercontent.com/5884000/34630456-783ac6a2-f26c-11e7-9bb7-79792288e364.png)
pls. change to 1 day

thx.

**Comment by jspenguin2017** [comment]
It is 1 day in Nano... which is what you use, right?

**Comment by gorhill** [comment]
@jspenguin2017 If you find this acceptable, good for you, please do not have your fork pull resources from here. Thank.

**Comment by KonoromiHimaries** [comment]
@gorhill you can use p2p : C

**Comment by jspenguin2017** [comment]
If a filter maintainer thinks it's appropriate, there is no reason to disallow 1 day expire time. He will deal with hosting problems himself if it turns out 1 day is too short. You are never forced to make uBlock filters update every day.

**Comment by jspenguin2017** [comment]
There are so many ways to save bandwidth: 
- Compress the file that you distribute (remove comments, combine rules, etc)
- Compress filter syntax, `xhr` for `xmlhttprequest`, `css` for `stylesheet`, etc
- Differential update
- Pack fresh filters in the extension and save an update

Instead, you choose to give users outdated filters while complaining about bandwidth. You can *easily* save a good 30% of bandwidth just by removing comments from the filter you distribute.

**Comment by gorhill** [comment]
@jspenguin2017 Sure, I can go on any GitHub repo, including yours, and throw lots of ideas about how things could be better and question "why didn't you do this?", having ideas is the easy part.

It's an ongoing project -- you seem to have this strange idea that everything you can muster in your mind _now_ should have been implemented in the past (on top of everything which was actually done), so that no issue whatsoever, small or big, should be afflicting the project now.

The bandwidth issue has been on my mind lately so as to not abuse GitHub, and it's not because I do not throw ideas for all to read that is a sign I need to be educated about what can be done.

**Comment by jspenguin2017** [comment]
I'm not saying you should've have all that implemented, I'm saying you should think about doing that instead of sticking the `declined` label and call it a day. 

I've been compressing my filters a long time ago and I'll compress the rest soon.

**Comment by jspenguin2017** [comment]
38% smaller not counting the resource file which needs to be compressed in another way (uglify maybe?).

**Comment by KonoromiHimaries** [comment]
> 38% smaller not counting the resource file which needs to be compressed in another way (uglify maybe?).

2 days - 38% = 29.76 hours

**Comment by mikhaelkh** [comment]
@gorhill 
> The reason is to lower abusive behavior toward the servers which host the filter lists.

You respect `! Expires:` syntax, so if filter list set it to `1 day`, why override it? Nobody's asking you to set `1 day` to default.
> Nobody pays for that bandwidth

Maybe not directly, but they have analytics. If you're worried, you can ask GitHub how they feel about being resources hosting for such popular extension.

**Comment by thomasklash** [comment]
@gorhill does it again

**Comment by gorhill** [comment]
I figured I lost the argumentation, hence the fix.

**Comment by thomasklash** [comment]
> I figured I lost the argumentation, hence the fix.
Well yeah, but regardless of outcome, you are unnecessarily rude and belligerent on almost every bug or pull request that comes your way.

**Comment by gorhill** [comment]
> rude and belligerent

Can you point me to an example of this? I want to understand your perception.

**Comment by thomasklash** [comment]
> Can you point me to an example of this? I want to understand your perception.

I've watched this repository for a good year or so now, and its just the impression I get from your responses to bugs, and in particular, criticism. It's like a syndrome where a developer runs a project, open-source or otherwise, that garners a lot of attention, and then they become egotistical or neurotically protective of their work.

If you really want examples, I'll email you or something.

**Comment by seanrand** [comment]
@thomasklash see https://github.com/gorhill/uBlock/issues/2649#issuecomment-304310972 for some context. Community management/moderation can be taxing, it's nothing personal. I've seen devs abandon their projects because they didn't want to deal with the constant flow of nonsense once it reached a certain popularity. The signal-to-noise ratio tends to reach lows especially on general purpose projects that have lots of non-technical users.

**Comment by thomasklash** [comment]
@seanrand Yes, I know. I am involved in the Arch community and have seen this happen frequently: see the most recent developments with the now discontinued `pacaur` utility.

However, in that case, this case and many others, all of these non-technical users that flood in mean well, apart from some rudeness and some trolls. I know sometimes you need to give people a hard no, but too often I see people going the Torvalds route and just raining down on someone who is simply inexperienced, not that intelligent, or whatever.

The last thing the softdev/engineering/compsci/IT community and industry needs is more cynical assholes who have had their share of shit thrown at them and are now bitter and rude towards anyone not intimately familiar with the project, its scope, or just how things work in development general.

**Comment by Hrxn** [comment]
@thomasklash You're just projecting. Stop making up problems where none exist.

**Comment by mikhoul** [comment]
[**@thomasklash**](https://github.com/thomasklash) commented on [24 févr. 2018 à 02:09 UTC−5](https://github.com/gorhill/uBlock/issues/3406#issuecomment-368206647 "2018-02-24T07:09:07Z - Replied by Github Reply Comments"):
> > Can you point me to an example of this? I want to understand your perception.
> 
> I've watched this repository for a good year or so now, and its just the impression I get from your responses to bugs, and in particular, criticism. It's like a syndrome where a developer runs a project, open-source or otherwise, that garners a lot of attention, and then they become egotistical or neurotically protective of their work. I see it everywhere.
> 

I agree 200% with @thomasklash , I stopped to report bugs or make suggestion on UBO because of the attitude of the maintainer. Also I've been involved and I'm involved on MANY other Open source projects and from my personal experience  I never seen an attitude such bad as here when somebody make suggestions or ask for help.

I'm so happy for the fork of @jspenguin2017 since it's "fresh air" with an "open mind"...  It may not agree with all suggestion but you don't feel like Sh***it when he answer to you.  You feel that he realy care about the user,

@gorhill attitude is like Mozilla/Firefox attitude with their users..... they pretend know better than users what they need and want sadly and don't really listen unless it's the way they see things.

**N.B.:** This message is not to bash @gorhill, **I'm thank him for his contribution** but I hate his attitude and closed mind when he answer here on Github.

Regards :octocat:

**Comment by Hrxn** [comment]
You're the ones ruining open source projects. That is for sure.

**Comment by thewriteway** [comment]
I think its @thomasklash who is disrespectful to mount a personal attack on Gorhill. You also come here demanding features at the snap of a finger. That isn't how open source works. If you want something added that you feel is super important to you then work on the solution yourself and contribute it back for others if you are feeling so in kind.  Gorhill does his best to make the project the best possible and I don't think you are in a position to grand stand and make accusations when you haven't (I'm assuming) committed anything to this project? Just be grateful uBlock even exists, I know I am and I wouldn't rag on the developers for things you consider important, because at the end of the day its their time being spent doing things not yours!!

**Comment by mikhaelkh** [comment]
I am amazed how @gorhill single-handedly maintains a project with >10M users. This is not a question of rudeness but rather a question of resources. The author simply does not have the time to explain every decision and sometimes like in this issue he forgot to check it after he closed it, that's unfortunate, not rude.

**Comment by tomasz1986** [comment]
I think that one of the problems may be that software developers (and computer savvy people in general) are very far from reality of "normal" computer users. Most people have completely no idea about how programs are written (open source or not), or how they work. They just see that something is wrong, and look for a way to report their problem to the program's author. They may have never heard of GitHub before, they do not know how it works, and the bug report may be the first one in their lifetime.

When it comes to uBlock, I really doubt that the majority of its users really understand the distinction between the program itself and the filter lists. So they come here and report their issues. Then they are greeted with a "declined" label with a short note that they are unable to understand. The result often is that they simply get angry and bash the software and its author. The more popular a piece of software is, the more such inexperienced users will use it (and come here with their problems).

I am not bashing or complaining, but maybe a more detailed explanation (even if just copied & pasted) would simply help those users understand better why their bug report has been declined, and let them know what they should do next?

**Comment by Hrxn** [comment]
I hope this is some sort of joke...
Sorry, but if you are "computer savvy" enough to know this site (GitHub), know how to create an account on this site, know how to log in and use said account, know how to navigate to the repository home page,
know that you have to click on "Issues", and know that you then can click on "New Issue" to open a new entry on here, _then_ it is definitely not possible to explain how you missed the text _right in front of you_, that clearly says that you should read `CONTRIBUTING.md` first, and that info box now always displayed by GitHub that also tells you to _"Take a look before submitting an issue: ... "_ and directly links the [Contributing guidelines](https://github.com/gorhill/uBlock/blob/master/CONTRIBUTING.md) for you, you're running out of excuses pretty quick. And don't let it be laziness and/or ignorance, please..

**Comment by jspenguin2017** [comment]
@Hrxn That is exactly my expectation, especially for other developers, but as I learnt from my experience, this is not reasonable to expect. In earlier 2017, Adguard devs basically put a giant middle finger towards my previous (now deleted) contribution guideline in their app, and it's how the silly fight happened as I expected them to read my contribution guideline. 

Expecting people to read and follow your contribution guideline will end in conflict. Having a guideline is good, but expecting that you can peacefully enforce it is not.

**Comment by thomasklash** [comment]
When have I ever demanded a feature lmfao, get a grip mate

On 25 Feb. 2018 04:57, thewriteway <notifications@github.com> wrote:

I think its @thomasklash<https://github.com/thomasklash> who is disrespectful to mount a personal attack on Gorhill. You also come here demanding features at the snap of a finger. That isn't how open source works. If you want something added that you feel is super important to you then work on the solution yourself and contribute it back for others if you are feeling so in kind. Gorhill does his best to make the project the best possible and I don't think you are in a position to grand stand and make accusations when you haven't (I'm assuming) committed anything to this project? Just be grateful uBlock even exists, I know I am and I wouldn't rag on the developers for things you consider important, because at the end of the day its their time being spent doing things not yours!!

**Comment by jspenguin2017** [comment]
I usually avoid controversial issues, but I'll put my 2 cents in...

First, I'm neutral on how gorhill is handling moderation, although I agree that his saved replied should be improved (which he did as far as I see), I don't agree that he should always argue with you and explain everything. On one side, it is hard to imagine not understanding a concept, on the other hand, developers are not teachers and have a hard time passing the knowledge over in an understandable way. It is totally OK to leave an issue closed or to lock it if a decision is made. (I think a direct link to Reddit in the popup menu should help in reducing noise in issues tracker though.)

However, looking back, a project of this scale is basically a one-man-show clearly indicates something is not right. It is fine to immediately close suggestions that are ridiculous in the common sense of developers, but those that are hard to tell should be left open for further discussion. 

That being said, gorhill is the king in his repositories, open source does not imply democracy, it just means you can create your own kingdom by clicking Fork. Open source projects can live with just 1 user but can die even if it has thousands, the ultimate decision is up to the developer.

**Comment by tomasz1986** [comment]
@Hrxn
>I hope this is some sort of joke...
Sorry, but if you are "computer savvy" enough to know this site (GitHub), know how to create an account on this site, know how to log in and use said account, know how to navigate to the repository home page,
know that you have to click on "Issues", and know that you then can click on "New Issue" to open a new entry on here, then it is definitely not possible to explain how you missed the text right in front of you, that clearly says that you should read CONTRIBUTING.md first, and that info box now always displayed by GitHub that also tells you to "Take a look before submitting an issue: ... " and directly links the [Contributing guidelines](Contributing guidelines) for you, you're running out of excuses pretty quick. And don't let it be laziness and/or ignorance, please..

I did exaggerate a little 😁, but if you have in fact seen how non-tech savvy people deal with computers and software, then I would not be so sure about it...

From my experience, as someone who has helped hundreds of people with their computers, people never actually read any documentation unless absolutely necessary. This includes all kinds of readmes, licenses, and other similar documents (such as CONTRIBUTING.md too). If people have read them, then I am pretty sure that at least half of the reported "issues" would not have had existed in the first place. I am not saying that it should be like that. I am rather just stating the reality, especially since GitHub is not some kind of a closed mailing list for developers, but is an open platform where anyone can register and post.

Is is due to their laziness? Maybe. I actually believe that they simply do not consider the software and its issues important enough to spend more time on studying them. They just want a "quick fix".

**Comment by Raavizz** [comment]
> @Hrxn : @thomasklash You're just projecting. Stop making up problems where none exist.
> @thewriteway : I think its @thomasklash who is disrespectful to mount a personal attack on Gorhill

I agree. @thomasklash, @mikhoul, @tomasz1986 stop yours drama / off-topic / harassing / accusations / slanders / negativity.
